nucypher/policy/conditions/evm.py

```diff
@@ -51,11 +50,22 @@ def _resolve_abi(standard_contract_type: str, method: str, function_abi: List) -
 
     if standard_contract_type:
         try:
-            function_abi = STANDARD_ABIS[standard_contract_type]
+            contract_abi = STANDARD_ABIS[standard_contract_type]
         except KeyError:
             raise ReencryptionCondition.InvalidCondition(
                 f"Invalid standard contract type {standard_contract_type}; Must be one of {STANDARD_ABI_CONTRACT_TYPES}"
             )
+        functions = Web3().eth.contract(abi=contract_abi).find_functions_by_name(method)
+        if not functions:
+            raise ReencryptionCondition.InvalidCondition(
+                f"Invalid method {method} for standard contract type {standard_contract_type}"
+            )
+        if len(functions) > 1:
+            raise ReencryptionCondition.InvalidCondition(
+                f"Multiple functions with name {method} for standard contract type {standard_contract_type}"
+            )
+        function_abi = [func.abi for func in functions][0]
```

Comment:
I opened a PR (#1) into your fork <https://github.com/theref/nucypher/pull/1> that simplifies this function a bit more and adds some strict type annotations 🙏🏻